core/domain/stats_jobs_continuous.py

```diff
@@ -611,8 +611,9 @@ def reduce(key, stringified_values):
             # version is equal to the latest version of the exploration,
             # otherwise ignore all answers since none of them can be applied to
             # the latest version.
-            exp = exp_services.get_exploration_by_id(exploration_id)
-            if state_name in exp.states:
+            exp = exp_services.get_exploration_by_id(
+                exploration_id, strict=False)
+            if exp and state_name in exp.states:
```

Comment:
Consider just returning if we discover that exp is None -- seems more resilient to future change, since we can then assume from this point on that the exploration exists and is fully valid.

Also, maybe move this out of the "if" clause? Even if previous exp versions are available I don't see the point in computing any models for particular versions if the exploration has been deleted.

Reply:
Done.